### DISCUSSION TITLE
Contember Interface in Next.js

### DISCUSSION THREAD

**Discussion by h0byte** [contember/admin#537] [Support]
Hi,

I would like to ask, is it possible to use Contember Interface components in Next.js just as an npm package?

We might want to implement admin pages authentication/authorization using Auth.js.

Thank you kindly! :)

**Comment by honzasladek** [comment]
Hi @sammwashere first thank you for your interest in Contember! I'll answer a bit more broadly because I'm not sure of your use case, happy to answer all the follow up questions. 

Contember Interface is designed specifically to easily work with Contember Engine. It's basically just an React single page application so while all the necessary logic around authentication and authorization are done in [Tenant API](https://docs.contember.com/reference/engine/tenant/overview) in [Contember Engine](https://docs.contember.com/). Our tenant API (apart from standard login/password flow) supports major identity providers like Facebook or Apple ([here's documentation](https://docs.contember.com/reference/engine/tenant/idp)).

Contember Interface primarily solves the [data binding layer](https://docs.contember.com/reference/admin/data-binding/overview/) between Engine and Interface. So you just drop in a component like this: 
```
<Field field='category.name' />
```
and don't have to write any queries to your GraphQL API (provided by Contember Engine). Therefore it's super fast to create any interface you want.

Regarding the use with Next.js: Interface works as a standard framework so it has it's own routing (for one thing). With some work it should be possible to use with Next.js (and we're thinking of making is super easy). @matej21 says it shouldn't be hard on our side, not sure how possible it would currently be without forking the Interface. @matej21 can advice more.
